packages/design-system/src/interfaces/banner.d.ts

```diff
@@ -0,0 +1,12 @@
+export type BannerSize = 'small' | 'medium' |'large';
+export type BannerType = 'info' | 'warning' | 'success' | 'error'
+
+export interface IBannerProps extends React.HTMLAttributes<HTMLDivElement> {
+  closeIcon?: boolean;
+  onClose?(e: React.MouseEvent<HTMLButtonElement>): void;
```

Comment:
`e` should not be required. You can use different type here:
```suggestion
 onClose?: MouseEventHandler<HTMLButtonElement>
```